### PR TITLE
chore: Kotlin 1.8.22

### PR DESCRIPTION
**Pull request by SimonMarquis**
- Closes #436
- Closes #442

### CODE REVIEW

File: build-logic/src/main/kotlin/com/adevinta/spark/SparkAndroidComposePlugin.kt [hunk 1]
```diff
@@ -39,6 +41,14 @@ internal class SparkAndroidComposePlugin : Plugin<Project> {
                 }
             }
 
+            @Suppress("ktlint:max-line-length")
```
Comment (soulcramer) [inline build-logic/src/main/kotlin/com/adevinta/spark/SparkAndroidComposePlugin.kt:44]: Didn't we had this already 🤔 ?

**Review by soulcramer — DISMISSED** [review]

File: build-logic/src/main/kotlin/com/adevinta/spark/SparkAndroidComposePlugin.kt (hunk 1 above)
Comment (SimonMarquis) [reply build-logic/src/main/kotlin/com/adevinta/spark/SparkAndroidComposePlugin.kt:44]: The `ktlint` suppression or the `suppressKotlinVersionCompatibilityCheck`?

Comment (SimonMarquis) [reply build-logic/src/main/kotlin/com/adevinta/spark/SparkAndroidComposePlugin.kt:44]: this is not needed when the kotlin version matches the expected version requested by the compose compiler. And this is not the case here.
Compose compiler version 1.4.7 expects Kotlin 1.8.21.
This is not very important for such patch release, but we can wait until compose releases (if it ever does) a new 1.4.8 version.

Comment (soulcramer) [reply build-logic/src/main/kotlin/com/adevinta/spark/SparkAndroidComposePlugin.kt:44]: Okay good to know thanks

**Comment by SimonMarquis** [comment]
@soulcramer PTAL

**Review by soulcramer — APPROVED** [review]

**Pull request merged**